core/services/vrf/listener_v2.go

```diff
@@ -757,6 +787,18 @@ func (lsn *listenerV2) processRequestsPerSub(
 					ll.Criticalw("Pipeline error", "err", p.err)
 				} else {
 					ll.Errorw("Pipeline error", "err", p.err)
+
+					// Ensure consumer is valid, otherwise drop the request.
+					if !lsn.isConsumerValidAfterFinalityDepthElapsed(ctx, p.req) {
```

Comment:
does this block need to be outside the `else`? unless we somehow don't want to check consumer is valid when `errBlockhashNotInStore`

Reply:
The blockhash should be stored and if it isn't that's a separate issue. In this case, the blockhash error would give us an alert first, which we would resolve with backwards mode or something alike, then after the blockhash is stored the listener would see that the consumer is invalid. I suppose we can check sooner, it doesn't really make a difference either way from my pov.

Reply:
I kind of worry that these are two different errors that represent two different things, though. So maybe we should keep it separate.

Reply:
I'm in favor of handling blockhash error independently from invalid consumer so that blockhash error logs a critical warning then removes the invalid consumer log if needed. Otherwise, the removal of invalid consumer will be blocked. 

But the likelihood of both happening at the same time is very slim. So I'm ok with keeping it as is given that we also have a mitigation strategy - run backwards mode first